sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/AvroGenericRecordToStorageApiProtoTest.java

```diff
@@ -567,4 +579,30 @@ public void testMessageFromGenericRecordWithMap() throws Exception {
     }
     assertEquals(mapData, actualMap);
   }
+
+  @Test
+  public void testMessageFromGenericRecordWithNullableArray() throws Exception {
+    ImmutableList<String> aList = ImmutableList.of("one", "two", "red", "blue");
+    GenericRecord recordWithMap =
+        new GenericRecordBuilder(SCHEMA_WITH_NULLABLE_ARRAY).set("aNullableArray", aList).build();
```

Comment:
For completion, can we also include in this test a record with null value for the array?

Reply:
sure.

Reply:
It was good suggestion to add this test case, helped me catch an error case. Added a fix for it.